### PR TITLE
testing/openresty: new aport

### PR DESCRIPTION
**Pull request by tcely**
https://openresty.org/
A Fast and Scalable Web Platform by Extending NGINX with LuaJIT

### CODE REVIEW

**Comment by tcely** [comment]
Updated and squashed https://github.com/alpinelinux/aports/pull/3246

File: testing/openresty/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+license="BSD"
+options="!check"
+depends="!nginx"
+depends_dev="gd-dev geoip-dev libressl-dev libxslt-dev pcre-dev readline-dev zlib-dev"
+makedepends="$depends_dev linux-headers"
+install=""
+subpackages="$pkgname-dev"
+source="https://openresty.org/download/$pkgname-$pkgver.tar.gz"
+
+builddir="$srcdir/$pkgname-$pkgver"
```
Comment (TBK) [inline testing/openresty/APKBUILD]: Can be removed since it is the default path.

File: testing/openresty/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Contributor: tcely <tcely@users.noreply.github.com>
+# Maintainer:
+pkgname=openresty
+pkgver=1.13.6.1
+pkgrel=0
+pkgdesc="Scalable Web Platform by Extending NGINX with Lua"
+url="https://openresty.org/"
+arch="all"
+license="BSD"
+options="!check"
```
Comment (TBK) [inline testing/openresty/APKBUILD]: Please add a comment why check is disabled.

**Comment by fabled** [comment]
We already ship `nginx-mod-http-lua` and various other openresty modules via the main nginx package's submodules. Does this have something new?

**Comment by tcely** [comment]
@fabled The short answer is yes. This is a bundle that happens to include Nginx.

> OpenResty is a full-fledged web platform that integrates the standard Nginx core, LuaJIT, many carefully written Lua libraries. OpenResty is not an Nginx fork. It is just a software bundle.

You can learn more about what's included at: https://openresty.org/en/components.html

**Comment by fabled** [comment]
We generally prefer not to ship same software multiple times in different bundles. Could the nginx modules be included in the main nginx build, and package the lua modules individually? We in fact already have `lua-resty-http` and `lua-resty-dns` packages in community. We can finally have a meta package for `openresty` that pulls in everything in the bundle. Yes, this is more work, but it'll reduce code duplication and build times.

**Comment by tcely** [comment]
@fabled I broke out sub-packages for the modules the current `nginx` aport is pulling from `openresty` GitHub repositories.

I'm not sure if the current `nginx` build is suitable for use instead of the binary built with this bundle. I don't believe it's worth the time to figure it out as it could very well change in the next release.

My suggestion is to incorporate this aport and stop building `openresty` modules as a third-party addition to the `nginx` aport.

**Pull request closed without merging**